Mathlib/Analysis/SpecialFunctions/Trigonometric/Bounds.lean

```diff
@@ -229,4 +229,45 @@ theorem cos_le_one_div_sqrt_sq_add_one {x : ℝ} (hx1 : -(3 * π / 2) ≤ x) (hx
   · simp
   · exact (cos_lt_one_div_sqrt_sq_add_one hx1 hx2 hx3).le
 
+/-- All the intervals on which `sin` is increasing -/
+lemma sin_monotone (n : ℤ) : MonotoneOn sin (Icc (n * (2 * π) - π / 2) (n * (2 * π) + π / 2)) := by
+  apply monotoneOn_of_deriv_nonneg
+  · apply convex_Icc
+  · fun_prop
+  · fun_prop
+  · intro x m
+    simp only [interior_Icc, mem_Ioo, deriv_sin] at m ⊢
+    rw [← cos_sub_int_mul_two_pi _ n]
+    exact Real.cos_nonneg_of_mem_Icc ⟨by linarith, by linarith⟩
```

Comment:
Thoughts on putting these next to https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/Trigonometric/Basic.html#Real.monotoneOn_sin etc?
 e.g.

```suggestion
lemma monotoneOn_sin' (n : ℤ) :
    MonotoneOn sin (Icc (n * (2 * π) - π / 2) (n * (2 * π) + π / 2)) := by
  intro a ha b hb hab
  rw [← Real.sin_sub_int_mul_two_pi a n, ← Real.sin_sub_int_mul_two_pi b n]
  apply Real.monotoneOn_sin <;> simp at * <;> (try constructor) <;> linarith
```

(naming is a bit awkward with the existing one)

Reply:
Agree this is better. Thank you!

Reply:
Will commit your change and then clean up.